airflow/api/common/mark_tasks.py

```diff
@@ -148,10 +148,10 @@ def set_state(
     qry_dag = get_all_dag_task_query(dag, session, state, task_id_map_index_list, dag_run_ids)
 
     if commit:
-        tis_altered = qry_dag.with_for_update().all()
+        tis_altered = session.execute(qry_dag.with_for_update()).scalars().all()
         if sub_dag_run_ids:
             qry_sub_dag = all_subdag_tasks_query(sub_dag_run_ids, session, state, confirmed_dates)
-            tis_altered += qry_sub_dag.with_for_update().all()
+            tis_altered += session.execute(qry_sub_dag.with_for_update()).scalars().all()
```

Comment:
```suggestion
            tis_altered += session.scalars(qry_sub_dag.with_for_update()).all()
```

I’m not quite sure but the docs seem to suggest this is superior…?

https://docs.sqlalchemy.org/en/20/orm/queryguide/select.html#selecting-orm-entities

> This is most easily achieved by using the Session.scalars() method to execute, rather than the Session.execute()

Reply:
Used session.scalars where possible https://github.com/apache/airflow/pull/31569/commits/37edb980f31fbe13e0de14ede679519cb5ec90fc